ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -9542,15 +9540,51 @@ void HqlCppTranslator::buildCsvListFunc(BuildCtx & classctx, const char * func,
     funcctx.addReturn(queryQuotedNullExpr());
 }
 
-static void expandDefaultString(StringBuffer & out, IHqlExpression * property, const char * defaultValue)
+static void expandDefaultString(StringBuffer & out, IHqlExpression * value, const char * defaultValue, IAtom * encoding)
 {
-    IHqlExpression * value = property ? property->queryChild(0) : NULL;
+    //If there are multiple alternatives use the first in the list as the default
+    if (value && value->getOperator() == no_list)
+        value = value->queryChild(0);
     if (value && value->queryValue())
-        value->queryValue()->getStringValue(out);
+    {
+        if (encoding == unicodeAtom)
+            getUTF8Value(out, value);
+        else
+            value->queryValue()->getStringValue(out);
+    }
     else
         out.append(defaultValue);
 }
 
+static IHqlExpression * forceToCorrectEncoding(IHqlExpression * expr, IAtom * encoding)
+{
+    //This is ugly.  Really it should cast to a varutf8 type - but that isn't implemented.  So instead it
+    //casts it to a utf8, type transfers it to a string, and then casts that to a varstring!
+    //Reimplement if varutf8 is ever implemented.
+    if (expr && (encoding == unicodeAtom))
+    {
+        if (expr->isList())
+        {
+            assertex(expr->getOperator() == no_list);
+            HqlExprArray args;
+            ForEachChild(i, expr)
+            {
+                IHqlExpression * value = expr->queryChild(i);
+                args.append(*forceToCorrectEncoding(value, encoding));
+            }
+            return expr->clone(args);
+        }
+        else
+        {
+            OwnedHqlExpr cast = ensureExprType(expr, unknownUtf8Type);
```

Comment:
What's unknownUtf8Type?  Not sure I understand how UTF-8 type can be an unknown type.  Should this be unknownUnicodeType?